piv/piv.go

```diff
@@ -476,19 +517,48 @@ func (yk *YubiKey) SetManagementKey(oldKey, newKey [24]byte) error {
 
 // ykSetManagementKey updates the management key to a new key. This requires
 // authenticating with the existing management key.
-func ykSetManagementKey(tx *scTx, key [24]byte, touch bool) error {
+func ykSetManagementKey(tx *scTx, key []byte, touch bool) error {
+	var managementKeyType byte
```

Comment:
This implicit key type switching based on length makes it very easy to use this API incorrectly. I'd like to see either:

a) an opinionated choice to keep the 24-byte key length and using AES192 for keys that support it and only 3DES for older keys without AES support. This is supported by the fact that Yubikey default algorithm is AES192 for newer keys; or

b) an explicit selection of key type exposed in the `SetManagementKey` function signature. Though this increases the API surface and I don't know of any scenarios where it would be useful to select AES128 or AES256 instead of AES192.

Reply:
a) I already do this, if it supports AES192 and a 24 byte key is passed, it uses AES192, not 3DES.

b) You pass it 16, 24, or 32 bytes (128, 192, or 256 bits). You _are_ explicitly telling it what you want by the length of key.

Reply:
Updated SetManagementKey to guard AES management key types behind an explicit version check.